commit 74511bb340059be5a3fceb032213c7f325344694
Author: Jens Taprogge <jens.taprogge@taprogge.org>
Date:   Sun Oct 26 18:16:09 2008 +0100

    USB: Unusual dev for the "Kyocera / Contax SL300R T*" digital camera.
    
    The camera reports an incorrect size and fails to handle PREVENT-ALLOW
    MEDIUM REMOVAL commands.  The patch marks the camera as an unusual dev
    and adds the flags to enable the workarounds for both shortcomings.
    
    Signed-off-by: Jens Taprogge <jens.taprogge@taprogge.org>
    Cc: Alan Stern <stern@rowland.harvard.edu>
    Cc: Phil Dibowitz <phil@ipom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/drivers/usb/storage/unusual_devs.h linux-source-2.6.26/drivers/usb/storage/unusual_devs.h
--- linux-source-2.6.26.orig/drivers/usb/storage/unusual_devs.h	2009-02-07 16:43:12.000000000 -0700
+++ linux-source-2.6.26/drivers/usb/storage/unusual_devs.h	2009-03-08 22:10:12.000000000 -0600
@@ -307,6 +307,13 @@ UNUSUAL_DEV(  0x0482, 0x0103, 0x0100, 0x
 		"Finecam S5",
 		US_SC_DEVICE, US_PR_DEVICE, NULL, US_FL_FIX_INQUIRY),
 
+/* Patch submitted by Jens Taprogge <jens.taprogge@taprogge.org> */
+UNUSUAL_DEV(  0x0482, 0x0107, 0x0100, 0x0100,
+		"Kyocera",
+		"CONTAX SL300R T*",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY | US_FL_NOT_LOCKABLE),
+
 /* Reported by Paul Stewart <stewart@wetlogic.net>
  * This entry is needed because the device reports Sub=ff */
 UNUSUAL_DEV(  0x04a4, 0x0004, 0x0001, 0x0001,
